base/linalg/lapack.jl

```diff
@@ -1205,8 +1205,8 @@ for (gelsd, gelsy, elty) in
             end
             newB = [B; zeros($elty, max(0, n - size(B, 1)), size(B, 2))]
             s     = similar(A, $elty, min(m, n))
-            rcond = convert($elty, rcond)
-            rnk   = Vector{BlasInt}(1)
+            rcond = Ref{$elty}(rcond)
+            rnk   = Ref{BlasInt}()
             info  = Ref{BlasInt}()
             work  = Vector{$elty}(1)
             lwork = BlasInt(-1)
```

Comment:
`lwork` is only an input parameter. I think it is cleaner only to make the output parameters `Ref`s.